spring-cloud-contract-verifier/src/main/groovy/org/springframework/cloud/contract/verifier/builder/XmlBodyVerificationBuilder.java

```diff
@@ -43,15 +44,20 @@ void addXmlResponseBodyCheck(BlockBuilder blockBuilder, Object responseBody,
 	}
 
 	private void addXmlProcessingLines(final BlockBuilder blockBuilder,
-			String responseString) {
-		Arrays.asList(
-				"DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()",
-				"Document parsedXml = documentBuilder.parse(new InputSource(new StringReader("
-						+ responseString + ")))")
-				.forEach(it -> {
-					blockBuilder.addLine(it);
-					addColonIfRequired(lineSuffix, blockBuilder);
-				});
+			SingleMethodBuilder methodBuilder, String responseString) {
+		methodBuilder.variable("documentBuilder", "DocumentBuilder");
+		blockBuilder
+				.appendWithSpace(
+						"= DocumentBuilderFactory.newInstance().newDocumentBuilder()")
+				.addEndingIfNotPresent().addEmptyLine();
+		methodBuilder.variable("parsedXml", "Document");
+		String value = KotlinClassMetaData.hasKotlinSupport()
```

Comment:
Here also I would rather have a Kotlin-specific implementation, Java-specific implementations, than conditions at this level.

Reply:
Fixed